gen/utils/data/data.go

```diff
@@ -0,0 +1,144 @@
+	// If it is not set, use the default host defined in constPubhubDevnetHost.
+	host := os.Getenv(constEnvMetaHost)
+	if host == "" {
+		host = constPubhubDevnetHost
+	}
+	ds.metaHost = host
+	genLogger.Info(fmt.Sprintf("Meta data host set to: %s.", host))
+}
+
+func (ds *DataStore) retrieveEnvMetaClassesFromRemote() {
+	// Retrieve the meta data for the classes specified in the constEnvMetaClasses environment variable.
+	classNames := strings.Split(os.Getenv(constEnvMetaClasses), ",")
+
+	// Avoid attempting retrieval for unset or set to empty ("") environment variable.
+	if classNames[0] != "" {
+		genLogger.Debug(fmt.Sprintf("Retrieving meta files for classes: %s.", classNames))
+		for _, className := range classNames {
+			// ENHANCEMENT: Concurrently retrieve/write the meta data.
+			// Retrieve the meta file for the class from the remote location.
+			// Only retrieve the meta file if it is not already retrieved.
+			if !slices.Contains(ds.retrievedClasses, className) {
+				ds.retrieveMetaFileFromRemote(className)
+			}
+		}
+	}
+}
+
+func (ds *DataStore) refreshMetaFiles() {
+	// Check if the meta data should be refreshed from remote location as specified in the environment variable.
+	metaRefresh := os.Getenv(constEnvMetaRefresh)
+	// If the environment variable is not set, the default is to not refresh the meta data.
+	if metaRefresh != "" {
+		var refresh bool
+		var err error
+		// If the environment variable is set, parse it to a boolean value.
+		// If the parsing fails, log a warning and skip retrieval of meta data.
+		// Parsing accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
+		// Any other value returns an error, which is logged and the retrieval is skipped.
+		refresh, err = strconv.ParseBool(metaRefresh)
+		if err != nil {
+			genLogger.Warn(fmt.Sprintf("Refreshing of meta is skipped due to error: %s.", err.Error()))
+		} else if refresh {
+			genLogger.Debug(fmt.Sprintf("Refreshing meta data from remote location: %s.", ds.metaHost))
+			for _, className := range utils.GetFileNamesFromDirectory(constMetaPath, true) {
+				// ENHANCEMENT: Concurrently retrieve/write the meta data.
+				// Retrieve the meta file for the class from the remote location.
+				// Only retrieve the meta file if it is not already retrieved.
+				if !slices.Contains(ds.retrievedClasses, className) {
+					ds.retrieveMetaFileFromRemote(className)
+				}
+			}
+			genLogger.Debug(fmt.Sprintf("Successfully refreshed meta data from remote location: %s.", ds.metaHost))
+		}
+
+	}
+}
+
+func (ds *DataStore) retrieveMetaFileFromRemote(className string) {
+	packageName, shortName := splitClassNameToPackageNameAndShortName(className)
+	url := fmt.Sprintf(constMetaFileUrl, ds.metaHost, packageName, shortName)
+	genLogger.Debug(fmt.Sprintf("Retrieving meta data for class '%s' from: %s.", className, url))
+
+	client := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
```

Comment:
Do we need to initialize a new client for every API call? Shouldn't we store it in the DataStore itself?

Reply:
will change